tests/testthat/test-concRespCore.R

```diff
@@ -441,6 +441,79 @@ test_that("HTTr signature data internal check", {
 
 })
 
+test_that("HTTr gene data internal check", {
 
+  skip_on_cran()
+
+  ## load necessary data
+  ## Code below is commented out, not needed when running all tests in the package at once (such as with testthat::test_local().)
+  ## Do need to un-comment and run this code to load the data if one is running this test interactively in the console.
+
+  #load(here::here("R", "sysdata.rda"))
+
+  # turn signature_input into a list of rows for lapply to use
+  gene_input = as.list(as.data.frame(t(gene_input), stringsAsFactors = F))
+
+  # run curve-fitting
+  my_gene = lapply(X=gene_input, FUN = concRespCore,
```

Comment:
@brown-jason: I may be misunderstanding the true intent here on Unit testing for R. But the concRespCore function seems to take in a lot of default values. For unit testing, there should be an exhaustive listing of tests for all the permutations. That inventory/listing could mark out which combinations are valid/invalid and worth testing. That being said, I am seeing a really large number that would be n! combinations and a bit ridiculous. Prioritizing what should be tested for someone like me would be reasonable.

The PR could document why some test permutations are better to test over others. I would also want to know why bmr_scale = 1.349 is hardcoded in the function definition. That seems like a possible error. (I know devs will hard code values to test, then forget to remove them later. I don't know enough to recognize if this is a special number like Avogadro's const, etc.)

I guess my question is what would happen if an element of required input data were missing? Would the code call out a problem with the inputs and stop processing? Log the error with instructional text? Most of these tests seems like they test the happy path through the code without showing me what exception handling gets triggered if the input data was not well-scrubbed.

Not sure if this is helpful or even applicable.